jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2484,16 +2485,24 @@ ProtectionDomain getProtectionDomainInternal() {
 	if (result != null) {
 		return result;
 	}
-	if (AllPermissionsPD == null) {
+	if (allPermissionsPD == null) {
 		allocateAllPermissionsPD();
 	}
-	return AllPermissionsPD;
+	return allPermissionsPD;
 }
 
 private void allocateAllPermissionsPD() {
-	Permissions collection = new Permissions();
-	collection.add(SecurityConstants.ALL_PERMISSION);
-	AllPermissionsPD = new ProtectionDomain(null, collection);
+	/* Synchronization to ensure safe initialization and safe publication of
+	 * allPermissionsPD. This addresses the locking contention on allPermissionsPD
+	 * in getProtectionDomain().
+	 */
+	synchronized(this) {
```

Comment:
Synchronization on a specific class doesn't prevent multiple threads from creating distinct `ProtectionDomain` objects. Why does this not use a singleton object (or `Class.class`) for synchronization?

Reply:
@keithc-ca You're correct. Since it is a static, we will need to lock on `Class.class`. Let me create a fix for it.

Reply:
Opened https://github.com/eclipse-openj9/openj9/pull/21077.

Let me try the above approach.